mgmt/src/processor/confbuild.rs

```diff
@@ -204,55 +204,56 @@ fn build_vpc_internal_config(
     Ok(())
 }
 
-/// Get the config for the (a) loopback interface
-fn get_loopback_interface_config(
-    external: &ExternalConfig,
-) -> Result<&InterfaceConfig, ConfigError> {
-    external
+struct VtepInfo {
+    ip: UnicastIpv4Addr,
+    mac: SourceMac,
+}
+
+fn get_vtep_info(external: &ExternalConfig) -> Result<VtepInfo, ConfigError> {
+    match external
         .underlay
         .vrf
         .interfaces
         .values()
-        .find(|config| matches!(config.iftype, InterfaceType::Loopback))
-        .ok_or(ConfigError::MissingParameter(
-            "loopback interface configuration",
-        ))
-}
-
-/// Get address (one of them) from loopback interface
-fn get_loopback_address(external: &ExternalConfig) -> Result<UnicastIpv4Addr, ConfigError> {
-    let lo = get_loopback_interface_config(external)?;
-    lo.addresses
-        .iter()
-        .find_map(|assignment| match assignment.address {
-            IpAddr::V4(ip) => {
-                if assignment.mask_len != 32
-                    || ip.is_loopback()
-                    || ip.is_broadcast()
-                    || ip.is_unspecified()
-                {
-                    return None;
-                }
-                UnicastIpv4Addr::new(ip).ok()
+        .filter(|config| matches!(config.iftype, InterfaceType::Vtep(_)))
+        .collect::<Vec<_>>()
+        .as_slice()
+    {
+        [] => Err(ConfigError::MissingParameter(
+            "Vtep interface configuration",
+        )),
+        [intf] => match &intf.iftype {
+            InterfaceType::Vtep(vtep) => {
+                let mac = match vtep.mac {
+                    Some(mac) => SourceMac::new(mac).map_err(|_| {
+                        ConfigError::BadVtepMacAddress(
+                            mac,
+                            "mac address is not a valid source mac address",
+                        )
+                    }),
+                    None => return Err(ConfigError::MissingParameter("VTEP MAC address")),
+                }?;
+
+                let ip = UnicastIpv4Addr::new(vtep.local).map_err(|_| {
+                    ConfigError::BadVtepLocalAddress(vtep.local.into(), "address is not unicast")
+                })?;
+                Ok(VtepInfo { ip, mac })
             }
-            IpAddr::V6(_) => None,
-        })
-        .ok_or(ConfigError::MissingParameter("loopback interface address"))
+            _ => unreachable!(),
+        },
+        _ => Err(ConfigError::TooManyInstances("Vtep interfaces", 1)),
+    }
 }
 
 /// Build the Vtep config
 fn build_vtep_config(external: &ExternalConfig, internal: &mut InternalConfig) -> ConfigResult {
     // TODO: The vtep configuration should come from the external config, but I am deriving it from
```

Comment:
I think these comments could be removed.

Reply:
Fixed.